samples/token_statistics/build/Jamfile.v2

```diff
@@ -19,7 +19,7 @@ exe token_statistics
     :
         $(SOURCES)
         /boost/wave//boost_wave
-        /boost/program_options//boost_program_options
+        /boost/program_options//boost_program_options/<link>static
```

Comment:
What's the rationale of this change? Could you elaborate, please?

Reply:
Sure! This stems from a problem linking on MSVC-12. A nearly identical issue was reported at https://github.com/boostorg/program_options/issues/58; after investigating it for a while I realized that the tactic used [here](https://github.com/boostorg/wave/blob/18f834ac355d65d18619af673122c23623e7c912/test/build/Jamfile.v2#L93) would fix this as well.

I did also try adding `BOOST_ALL_DYN_LINK` but then it seemed like some inappropriate things got marked `__dllimport`, in particular `expression_grammar_gen<TokenT>::evaluate` and MSVC didn't like that either.

I haven't tried using `BOOST_PROGRAM_OPTIONS_DYN_LINK` but could attempt it if you prefer that approach.

Reply:
Don't worry about this too much, I was just curious.